File: PolygonCalculatorUsingJS/web/Polygon.js
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
function FullCalculation(){  
    var n = parseInt(document.getElementById("sides").value);
    var radius = parseFloat(document.getElementById("radius").value);
    var side = (2 * radius * Math.sin(Math.PI/n));    
    var perimeter = (n * side);
    var area = (0.25 * n * side * side / Math.tan(Math.PI/n));
    document.getElementById("sidelength").value = side.toFixed(2);
    document.getElementById("perimeter").value = perimeter.toFixed(2);
    document.getElementById("area").value = area.toFixed(2);
}

window.onload = function() {
    document.getElementById("btnCalculate").onclick = FullCalculation;
    document.getElementById("sides").focus();
    
};

File: PolygonCalculatorUsingJSF/src/java/JSFPackage/Polygon.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package JSFPackage;

/**
 *
 * @author Ronak Bakshi
 */
import javax.inject.Named;
import javax.enterprise.context.RequestScoped;

@Named(value = "polygon")
@RequestScoped
public class Polygon {
        int numberOfSides;
        int radius;
        
    
    public Polygon() {
    }
    public int getNumberOfSides() {
        return numberOfSides;
    }
    public int getRadius() {
        return radius;
    }
    public void setNumberOfSides(int numberOfSides) {
        this.numberOfSides = numberOfSides;
    }
    public void setRadius(int radius) {
        this.radius = radius;
    }    
    public double getSideLength(){
        return 2 * radius * Math.sin(Math.PI/numberOfSides);
    }
    public double getPerimeter(){
        return numberOfSides * getSideLength();
    }
    public double getArea(){
        return 0.25 * numberOfSides * getSideLength() * getSideLength() / Math.tan(Math.PI/numberOfSides);
    }
}